pkg/app/application.go

```diff
@@ -36,31 +36,17 @@ import (
 	"github.com/pkg/errors"
 )
 
-// Application interface exposes methods to get resources of the application.
-type Application interface {
-	// Resources computes and returns the resources for the app.
-	Resources() (resmap.ResMap, error)
-	// SemiResources computes and returns the resources without name hash and name reference for the app
-	SemiResources() (resmap.ResMap, error)
-	// RawResources computes and returns the raw resources from the kustomization file.
-	// It contains resources from
-	// 1) untransformed resources from current kustomization file
-	// 2) transformed resources from sub packages
-	RawResources() (resmap.ResMap, error)
-	// Vars returns all the variables defined by the app
-	Vars() ([]types.Var, error)
-}
-
-var _ Application = &applicationImpl{}
-
-// Private implementation of the Application interface
-type applicationImpl struct {
+// Application implements the guts of the kustomize 'build' command.
+// TODO: Change name, as "application" is overloaded and somewhat
+// misleading (one can customize an RBAC policy).  Perhaps "Target"
+// https://github.com/kubernetes-sigs/kustomize/blob/master/docs/glossary.md#target
+type Application struct {
```

Comment:
I don't have preference of using `interface` over `struct` here since we only use `Application` in this package and I don't see any evident usage outside `kustomize`.

Reply:
Yeah, there was no use of this interface.  No multiple implementations, no use in a test where a mock of the interface could be used.  The struct is too "high" in the stack, it's basically the main, so the interface was just blocking me.